databricks/sdk/azure.py

```diff
@@ -15,19 +15,15 @@ class AzureEnvironment:
 ARM_DATABRICKS_RESOURCE_ID = "2ff814a6-3304-4ab8-85cb-cd0e6f879c1d"
 
 ENVIRONMENTS = dict(
-    PUBLIC=AzureEnvironment(name="AzurePublicCloud",
+    PUBLIC=AzureEnvironment(name="PUBLIC",
```

Comment:
From what I see, the name is not really used. Correct?

Reply:
In general, users don't need to specify this if they specify the host name, since we can derive the Azure environment from the hostname. However, if users only specify the workspace resource ID and are deploying in either China or Govcloud, they need to specify either `ARM_ENVIRONMENT` or the `azure_environment` parameter in WorkspaceClient/AccountClient/Config.

This PR does change this (before you would specify `AzurePublicCloud` and now you would specify `PUBLIC` or `public`), aligning with the [Azure TF provider](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#environment) and with the [Go SDK](https://github.com/databricks/databricks-sdk-go/blob/main/config/azure.go#L123).

Reply:
Follow-up: I added back compatibility for the old style of environment names just in case, so customers specifying these parameters now will not need to change.